# Oasis Infobyte SIP — Data Science Internship
## Task 5: Sales Prediction Using Python

**Author:** Poornima  
**Track:** Data Science  
**Task Title:** Task 5: Sales Prediction Using Python  
**GitHub Repository:** `OIBSIP`  

---


In [1]:
import os
import pandas as pd, numpy as np
import matplotlib.pyplot as plt, seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_absolute_error, r2_score

data_path = "Advertising.csv"
if not os.path.exists(data_path): data_path = "Advertising Budget and Sales.csv"
df = pd.read_csv(data_path)
df.columns = df.columns.str.strip()

column_mapping = {}
for col in df.columns:
    if 'TV' in col: column_mapping[col] = 'TV'
    elif 'Radio' in col: column_mapping[col] = 'Radio'
    elif 'Newspaper' in col: column_mapping[col] = 'Newspaper'
    elif 'Sales' in col: column_mapping[col] = 'Sales'

df.rename(columns=column_mapping, inplace=True)
df = df[[col for col in ['TV', 'Radio', 'Newspaper', 'Sales'] if col in df.columns]].copy()

display(df.head())


,TV,Radio,Newspaper,Sales
0,230.1,37.8,69.2,22.1
1,44.5,39.3,45.1,10.4
2,17.2,45.9,69.3,9.3
3,151.5,41.3,58.5,18.5
4,180.8,10.8,58.4,12.9


In [2]:
X = df[['TV', 'Radio', 'Newspaper']]
y = df['Sales']
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.20, random_state=42)

rf = RandomForestRegressor(n_estimators=100, random_state=42)
rf.fit(X_train, y_train)
rf_preds = rf.predict(X_test)

print("Random Forest R2 Score:", r2_score(y_test, rf_preds))
print("Random Forest MAE:", mean_absolute_error(y_test, rf_preds))


Random Forest R2 Score: 0.9813987063814621
Random Forest MAE: 0.6199249999999991


### Conclusion
Random Forest Regressor achieved an R2 score of over 98% in predicting sales revenue.